pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -91,17 +92,33 @@ func (a *v1beta1PolicyApplier) setAuthnFilterForPeerAuthn(proxyType model.NodeTy
 	p := config.Policy
 	p.Peers = []*authn_alpha.PeerAuthenticationMethod{}
 
-	effectiveMTLSMode := a.getMutualTLSModeForPort(port)
-	if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
-		mode := authn_alpha.MutualTls_PERMISSIVE
-		if effectiveMTLSMode == model.MTLSStrict {
-			mode = authn_alpha.MutualTls_STRICT
+	if proxyType == model.SidecarProxy {
+		effectiveMTLSMode := a.getMutualTLSModeForPort(port)
+		if effectiveMTLSMode == model.MTLSPermissive || effectiveMTLSMode == model.MTLSStrict {
+			mode := authn_alpha.MutualTls_PERMISSIVE
+			if effectiveMTLSMode == model.MTLSStrict {
+				mode = authn_alpha.MutualTls_STRICT
+			}
+			p.Peers = []*authn_alpha.PeerAuthenticationMethod{
+				{
+					Params: &authn_alpha.PeerAuthenticationMethod_Mtls{
+						Mtls: &authn_alpha.MutualTls{
+							Mode: mode,
+						},
+					},
+				},
+			}
 		}
+	} else {
+		// this is for gateway with a server whose TLS mode is ISTIO_MUTUAL
+		// this is effectively the same as strict mode. We dont really
```

Comment:
Also is it possible to add an e2e test for this?

Reply:
how? in multicluster? We dont even have a full gateway e2e in our system :(. Yes really no gateway e2e

Reply:
not necessarily multicluster, something like https://github.com/istio/istio/blob/86e571e0189f1b5be67d68a509f0df1488439db1/tests/integration/security/authorization_test.go#L430 should be enough, so basically you can enable mTLS between sidecar and ingressgateway using istio issued certs, and then apply an authorization policy on ingress gateway to check the principal field (both positive and negative cases), if the authn filter is added correctly, the authz policy should work.

Reply:
Or this one https://github.com/istio/istio/blob/118cd731a082d50266235185b8bca47c83e5a30c/tests/integration/security/jwt_test.go#L277